### ISSUE TITLE
Only one server per service is identified while using consulCatalog.

### ISSUE THREAD

**Issue by saideeptalari** [containous/traefik#5850]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Tried using the latest v2.1.0-rc2 and consulCatalog as a provider. I've defined the `config.toml` file as below.

```toml
[api]
    insecure = true
    debug = true

[providers.consulCatalog]
    exposedByDefault = false    
    [providers.consulCatalog.endpoint]
        address = "192.168.1.200:8501"
        scheme = "https"
    [providers.consulCatalog.endpoint.tls]
        ca = "<path-to-ca>"
        cert = "<path-to-cert>"
        key = "<path-to-key>"
        insecureSkipVerify = true

```

and spawned three containers with the following consul service definition for each.
```json
{
    "service": {
        "name": "web",
        "tags": [
            "http",
            "httpd",
            "testing",
            "traefik.enable=true",
            "traefik.http.routers.web.rule=Host(`web.localhost`)"
        ],
        "port": 80,
        "checks": [
            {
                "name": "HTTPD Check",
                "http": "http://127.0.0.1/",
                "method": "GET",
                "interval": "5s"
            }
        ]
    }
}
```

I'm able to see those three nodes in consul dashboard.
![image](https://user-images.githubusercontent.com/11706429/68989353-6b1ff200-086b-11ea-87d9-625353bc05c5.png)

In traefik dashboard only one server is shown instead.
![image](https://user-images.githubusercontent.com/11706429/68989360-986ca000-086b-11ea-9293-10878e644d6c.png)

### What did you expect to see?

All the three nodes should be registered.

### What did you see instead?

Only one node being registered as a server.

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
Version:      2.1.0-rc2
Codename:     cantal
Go version:   go1.13.4
Built:        2019-11-15T19:43:27Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Provided above.
```bash
sudo ./traefik --configFile config.toml
```

**Comment by mmatur** [comment]
Hi @saideeptalari,

Thanks for the interest on the project.

Unfortunately I am not able to reproduce your issue. Could you tell me more about:
- the number of consul agent in your cluster
- consul version
- the curl url you used to register your service
- the json content of http://127.0.0.1:8080/api/rawdata

Thanks in advance.

**Comment by saideeptalari** [comment]
I'm able to consistently reproduce this issue.

**Consul**:
Version: v1.5.3
Number of consul servers: 1
Number of consul agents: 3 (in separate docker containers)

I didn't use any url to register my service in consul or in traefik. Those services got registered using consul agents. Find the agent configuration.

```json
{
    "datacenter": "<name>",
    "retry_join": ["<address>:8301"],
    "data_dir": "state",
    "encrypt": "<encrypt-token>"
}
```

**Output** (from /api/rawdata)
```json
{"routers":{"api@internal":{"entryPoints":["traefik"],"service":"api@internal","rule":"PathPrefix(`/api`)","priority":2147483646,"status":"enabled","using":["traefik"]},"consul@consulcatalog":{"service":"consul","rule":"Host(`consul`)","status":"enabled","using":["http","traefik"]},"dashboard@internal":{"entryPoints":["traefik"],"middlewares":["dashboard_redirect@internal","dashboard_stripprefix@internal"],"service":"dashboard@internal","rule":"PathPrefix(`/`)","priority":2147483645,"status":"enabled","using":["traefik"]},"web@consulcatalog":{"service":"web","rule":"Host(`web22.localhost`)","status":"enabled","using":["http","traefik"]}},"middlewares":{"dashboard_redirect@internal":{"redirectRegex":{"regex":"^(http:\\/\\/[^:]+(:\\d+)?)/$","replacement":"${1}/dashboard/","permanent":true},"status":"enabled","usedBy":["dashboard@internal"]},"dashboard_stripprefix@internal":{"stripPrefix":{"prefixes":["/dashboard/","/dashboard"]},"status":"enabled","usedBy":["dashboard@internal"]}},"services":{"api@internal":{"status":"enabled","usedBy":["api@internal"]},"consul@consulcatalog":{"loadBalancer":{"servers":[{"url":"http://192.168.1.200:8300"}],"passHostHeader":true},"status":"enabled","usedBy":["consul@consulcatalog"],"serverStatus":{"http://192.168.1.200:8300":"UP"}},"dashboard@internal":{"status":"enabled","usedBy":["dashboard@internal"]},"web@consulcatalog":{"loadBalancer":{"servers":[{"url":"http://172.18.0.2:80"}],"passHostHeader":true},"status":"enabled","usedBy":["web@consulcatalog"],"serverStatus":{"http://172.18.0.2:80":"UP"}}}}
```

**Comment by mmatur** [comment]
@saideeptalari Thanks for the explanation. I am no able to reproduce your issue. 

I am working on a fix.

**Comment by traefiker** [comment]
Closed by #5900.
